### PR TITLE
Fix: dev server routing when not using subpath

### PR DESCRIPTION
**Pull request by matthewp**
## Changes

- Fixes routing when not using subpath. 
- Less clever

## Testing

No, going to do that as a follow-up. Will add more thorough dev server tests surrounding this behavior.

## Docs

N/A

### CODE REVIEW

File: packages/astro/src/core/dev/index.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export class AstroDevServer {
     this.port = config.devOptions.port;
     this.origin = `http://localhost:${this.port}`;
     this.site = config.buildOptions.site ? new URL(config.buildOptions.site) : undefined;
-    this.pathname = this.site ? this.site.pathname + '/' : '/';
+    this.pathname = this.site ? this.site.pathname : '/';
```
Comment (matthewp) [inline packages/astro/src/core/dev/index.ts:76]: This was forcing to always end with `/` causing some `//` when including that in the `buildOptions.site` config.

Comment (drwpow) [inline packages/astro/src/core/dev/index.ts:76]: Tiny suggestion (not blocking the PR): what about `devRoot` or something that indicates it’s the base directory of the dev server? `this.pathname` may be confused for request pathname inside certain contexts (e.g. `this.pathname` alongside `pathname`)

Comment (matthewp) [reply packages/astro/src/core/dev/index.ts]: Good call, updated.

**Review by drwpow — APPROVED** [review]
LGTM!

File: packages/astro/src/core/dev/index.ts [hunk 2]
```diff
@@ -277,10 +277,22 @@ export class AstroDevServer {
...
-        next();
-        return;
+      let routePathname: string = pathname;
+      // If using a subpath, ensure that the user has included the pathname
+      // such as /blog in the URL.
+      if(this.devRoot !== '/') {
+        if(pathname.startsWith(this.devRoot)) {
+          // This includes the subpath, so strip off the subpath so that
+          // matchRoute finds this route.
+          routePathname = pathname.substr(this.devRoot.length) || '';
```
Comment (FredKSchott) [inline packages/astro/src/core/dev/index.ts:287]: mentioned in chat: we have utility functions already to add/remove trailing/leading slashes.

**Pull request merged**